packages/theme/modules/catalog/pages/product.vue

```diff
@@ -126,6 +131,18 @@ export default defineComponent({
       addTags([...tags, ...productTags]);
     });
 
+    onMounted(async () => {
+      if (!product.value) {
+        await fetchProduct();
+      }
+
+      const { data } = await query<ProductDetailsQuery>(getProductPriceBySkuGql, { sku: product.value.sku });
+
+      product.value = {
+        ...merge(product.value, data.products?.items?.[0] as Product),
+      };
```

Comment:
https://docs-lodash.com/v4/merge/

merge mutates its first argument. I think you're looking for

```suggestion
      product.value = merge({}, product.value, data.products?.items?.[0] as Product);
```

Reply:
(without the spread ... is fine, as it'll be reference-free aswell)

Reply:
spread is ok but what about the first argument `{}`? Can it be added?

You should theoretically be able to do
```
product.value = merge({}, product.value, data.products?.items?.[] as Product);
```
or
```
merge(product.value, data.products?.items?.[] as Product)
```

Reply:
Fixed.